src/ansi-c/c_expr.h

```diff
@@ -225,4 +225,98 @@ to_history_expr(const exprt &expr, const irep_idt &id)
   return ret;
 }
 
+/// \brief A class for an expression that represents a conditional target or
+/// a list of targets sharing a common condition
+/// in an assigns clause.
+class conditional_target_exprt : public exprt
+{
+public:
+  explicit conditional_target_exprt(exprt _condition, exprt _target_list)
+    : exprt(ID_conditional_target, empty_typet{})
+  {
+    add_to_operands(std::move(_condition));
+    add_to_operands(std::move(_target_list));
```

Comment:
Why did you choose to no longer make it a `binary_exprt`?

Reply:
I was't really sure what a `binary_exprt` is supposed to represent, but from reading the code it seemed to be meant for binary operators such as `<`, `+`, where both left and right subtrees are homogenous.
This one being asymmetric and only meant to occur at a specify place in the AST seemed like a special case, it did not seem like reusing the binary_exprt interface would bring much.

Reply:
Ok! (Adding/subtracting pointers is an existing example of `binary_exprt` with non-matching types.)